main.odin

```diff
@@ -1077,160 +1077,129 @@ write_doc_line :: proc(w: io.Writer, text: string) {
-	} else {
-		io.write_string(w, s)
-	}
-}
-
-write_docs :: proc(w: io.Writer, docs: string) {
-	if docs == "" {
-		return
-	}
-	Block_Kind :: enum {
-		Paragraph,
-		Code,
-	}
-	Block :: struct {
-		kind: Block_Kind,
-		lines: []string,
-	}
-
-	lines: [dynamic]string
-	it := docs
-	for line_ in strings.split_iterator(&it, "\n") {
-		line := strings.trim_right_space(line_)
-		append(&lines, line)
 	}
 
-	curr_block_kind := Block_Kind.Paragraph
-	start := 0
-	blocks: [dynamic]Block
-
-	for line, i in lines {
-		text := strings.trim_space(line)
-		switch curr_block_kind {
-		case .Paragraph:
-			if strings.has_prefix(line, "\t") {
-				if i-start > 0 {
-					append(&blocks, Block{curr_block_kind, lines[start:i]})
-				}
-				curr_block_kind, start = .Code, i
-			} else if text == "" {
-				if i-start > 0 {
-					append(&blocks, Block{curr_block_kind, lines[start:i]})
-				}
-				start = i
-			}
-		case .Code:
-			if text == "" || strings.has_prefix(line, "\t") {
-				continue
-			}
+	// Normal text
+	io.write_string(w, line^)
+	io.write_string(w, " ")
+}
 
-			if i-start > 0 {
-				append(&blocks, Block{curr_block_kind, lines[start:i]})
-			}
-			curr_block_kind, start = .Paragraph, i
-		}
-	}
-	if start < len(lines) {
-		if len(lines)-start > 0 {
-			append(&blocks, Block{curr_block_kind, lines[start:]})
+write_markup_text :: proc(w: io.Writer, lines: []string) {
+	io.write_string(w, "<span style=\"white-space: pre-wrap\">") // Preserve leading whitespace
+	io.write_string(w, "<p>")
+	
+	line_loop: for i := 0; i < len(lines); i += 1 {
+		line := lines[i]
+		trimmed_line := strings.trim_space(line)
+
+		// Paragraph break
+		if len(trimmed_line) == 0 {
+			io.write_string(w, "</p>")
+			io.write_string(w, "<p>")
+			continue
 		}
-	}
 
-	for block in &blocks {
-		trim_amount := 0
-		for trim_amount = 0; trim_amount < len(block.lines); trim_amount += 1 {
-			line := block.lines[trim_amount]
-			if strings.trim_space(line) != "" {
-				break
+		// Lists
+		if trimmed_line[0] == '-' {
+			io.write_string(w, "<ul>")
+			for len(trimmed_line) > 0 && trimmed_line[0] == '-' {
+				io.write_string(w, "<li>")
+				trimmed_line_to_write := trimmed_line[1:]
+				write_markup_single_line(w, &trimmed_line_to_write)
+				io.write_string(w, "</li>")
+				
+				i += 1;
+				if i >= len(lines) {
+					break line_loop
+				}
+				
+				line = lines[i]
+				trimmed_line = strings.trim_space(line)
 			}
-		}
-		block.lines = block.lines[trim_amount:]
-	}
-
-	for block, i in blocks {
-		if len(block.lines) == 0 {
+			io.write_string(w, "</ul>")
 			continue
 		}
-		prev_line := ""
-		if i > 0 {
-			prev_lines := blocks[i-1].lines
-			if len(prev_lines) > 0 {
-				prev_line = prev_lines[len(prev_lines)-1]
-			}
-		}
-		prev_line = strings.trim_space(prev_line)
 
-		lines := block.lines[:]
-
-		end_line := block.lines[len(lines)-1]
-		if block.kind == .Paragraph && i+1 < len(blocks) {
-			if strings.has_prefix(end_line, "Example:") && blocks[i+1].kind == .Code {
```

Comment:
Any reason you got rid of the `Example:` prefix functionality?

Reply:
I didn’t get rid of it completely, just changed how it works slightly so you can use whatever text you want as the summary, not just “Example:”. You can see how it’s works in the code block example i provided (it uses whatever text is on the line before the triple backticks).